### PR TITLE
Updated arch to alpha5 and fixed an issue with the TypeConverters.

### PR DESCRIPTION
**Pull request by carlosefonseca**
Newer versions need the two TypeConverter methods to have the same types to compile.
Took me a while to understand why it wouldn't compile, so I'm sending this PR to help someone else looking for this.

### CODE REVIEW

File: app/src/main/kotlin/com/generalmobile/app/gmnotes/db/converters/DateConverter.kt [hunk 1]
```diff
@@ -11,7 +11,7 @@ class DateConverter {
     }
 
     @TypeConverter
-    fun dateToTimestamp(date: Date?): Long {
-        return date?.time ?: 0L
+    fun dateToTimestamp(date: Date?): Long? {
```
Comment (dtunctuncer) [inline app/src/main/kotlin/com/generalmobile/app/gmnotes/db/converters/DateConverter.kt:14]: @carlosefonseca Why did you change this line ?

**Review by dtunctuncer — COMMENTED** [review]
@carlosefonseca Why did you change this line ?

File: app/src/main/kotlin/com/generalmobile/app/gmnotes/db/converters/DateConverter.kt (hunk 1 above)
Comment (carlosefonseca) [reply app/src/main/kotlin/com/generalmobile/app/gmnotes/db/converters/DateConverter.kt:14]: Apparently both methods much have the same corresponding types (one was `Long`, the other was `Long?`), otherwise the compiler would give an error about not being able to find the typeconverter for the type.
